src/proxy/socks5.rs

```diff
@@ -107,7 +107,7 @@ impl Socks5 {
                                 debug!(component="socks5", dur=?start.elapsed(), "connection completed");
                             }).instrument(span);
 
-                            assertions::size_between_ref(1000, 2000, &serve);
+                            assertions::size_between_ref(1000, 10000, &serve);
```

Comment:
was erroring at at 5600

Reply:
If we went from 2000 to 5600 we have a problem we need to address. The size here is the per-connection memory overhead, so increases here directly lead to increased memory utilization

Reply:
I looked into this a bit more and "solved" it by boxing the futures. However, I don't know if this makes a difference the resulting binary because the size of futures can change based on compiler optimizations. For example, if I leave `proxy_to_double_hbone` unboxed, and restore the limit to 2000, the tests will pass if I build them with `--release`, but fail otherwise. So the size of `serve` seems more like a red herring.

Reply:
Ok, I take this back. Not sure if the size of futures changes based on optimization level, but I do find it a little silly that we have async functions at always box a futures certain futures for example

https://github.com/istio/ztunnel/blob/dc54d850a774e033cc0aa36357f623b8141e0926/src/proxy/outbound.rs#L170-L175

this always runs, so what's the point of boxing it? the memory usage will always be the same, except that we are moving it from the stack to the heap. The only team I think this makes sense is that if a future is really big and is not in the happy path. @howardjohn